src/liballoc/arc.rs

```diff
@@ -169,7 +169,7 @@ impl<T> Arc<T> {
             weak: atomic::AtomicUsize::new(1),
             data: data,
         };
-        Arc { _ptr: unsafe { NonZero::new(mem::transmute(x)) } }
+        Arc { _ptr: unsafe { NonZero::new(mem::transmute(x)).unwrap() } }
```

Comment:
This should convert `Box<T>` to `Unique<T>` or at least `NonZero<*mut T>`, not to `*mut T`.